tracer/src/Datadog.Tracer.Native/clr_helpers.cpp

```diff
@@ -1018,7 +1018,318 @@ HRESULT HasAsyncStateMachineAttribute(const ComPtr<IMetaDataImport2>& metadataIm
         methodDefToken, WStr("System.Runtime.CompilerServices.AsyncStateMachineAttribute"), &ppData, &pcbData);
 
     IfFailRet(hr);
-    hasAsyncAttribute = pcbData > 0 ? true : false;
+    hasAsyncAttribute = pcbData > 0;
     return hr;
 }
+
+HRESULT IsByRefLike(const ComPtr<IMetaDataImport2>& metadataImport, const mdTypeDef typeDefToken, bool& hasByRefLikeAttribute)
+{
+    const void* ppData = nullptr;
+    ULONG pcbData = 0;
+    auto hr = metadataImport->GetCustomAttributeByName(
+        typeDefToken, WStr("System.Runtime.CompilerServices.IsByRefLikeAttribute"), &ppData, &pcbData);
+
+    IfFailRet(hr);
+    hasByRefLikeAttribute = pcbData > 0;
+    return hr;
+}
+
+HRESULT ResolveTypeInternal(ICorProfilerInfo4* info,
+                            const std::vector<ModuleID>& loadedModules,
+                            const std::vector<WCHAR>& refTypeName,
+                            const mdToken parentToken,
+                            const shared::WSTRING& resolutionScopeName, mdTypeDef& resolvedTypeDefToken,
+                            ComPtr<IMetaDataImport2>& resolvedTypeDefMetadataImport)
+{
+    mdAssembly candidateAssembly;
+    auto isModuleFound = false;
```

Comment:
```suggestion
    auto wasModuleFound = false;
```
or just `foundModule`

Reply:
Done in ef502336ac5073c545e222bc6396635a9d76288f.